modules/service/src/main/resources/db/migration/V0927__source_profile_summary.sql

```diff
@@ -0,0 +1,43 @@
+CREATE TYPE e_source_profile_type AS ENUM(
+  'point',
+  'uniform',
+  'gaussian'
+);
+
+CREATE TYPE source_profile_summary AS (
+  c_profile_type e_source_profile_type,
+  c_fwhm         d_angle_µas
+);
+
+CREATE OR REPLACE FUNCTION extract_source_profile_summary(
+  src_profile  jsonb
+)
+RETURNS source_profile_summary AS $$
```

Comment:
This needs testing but we would need to extract the source profile type (point, uniform, or gaussian) from the JSON blob along with the fwhm if gaussian.  This information is needed to compute the object size for the binning calculation.

Reply:
This is my fault … I was concerned that designing a table structure for source profiles would have driven me mad.